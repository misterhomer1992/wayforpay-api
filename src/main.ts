export { generatePurchaseUrl } from './purchase';
export { generateRegularPurchaseUrl } from './regularPurchase';
