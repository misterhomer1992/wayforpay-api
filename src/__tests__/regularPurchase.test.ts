jest.mock('axios');
import { generateRegularPurchaseUrl } from '../main';
import axios from 'axios';

const mockedAxios = jest.mocked(axios);

describe('test generateRegularPurchaseUrl', () => {
    beforeEach(() => {
        mockedAxios.post.mockResolvedValueOnce({
            data: {
                url: 'https://secure.wayforpay.com/page?vkh=regular-some-fake-url-123',
            },
        });
    });

    afterEach(() => {
        mockedAxios.post.mockClear();
    });

    it('should generate regular payment url with regularCount prop', async () => {
        const date = 1715267902885;
        const response = await generateRegularPurchaseUrl({
            merchantAccount: 'my_account',
            merchantDomainName: 'https://my_account.com',
            merchantSecretKey: '12jk3hkj12h3kj1h3kj1h3kj123kh12',
            orderDate: date,
            orderReference: `product__${date}`,
            productName: ['Apple'],
            productPrice: [100],
            currency: 'UAH',
            regularMode: 'daily',
            regularCount: 1,
        });

        expect(mockedAxios.post).toMatchSnapshot();
        expect(response.url).toBe(
            'https://secure.wayforpay.com/page?vkh=regular-some-fake-url-123',
        );
    });

    it('should generate regular payment url with dateEnd prop', async () => {
        const date = 1715267902885;
        const response = await generateRegularPurchaseUrl({
            merchantAccount: 'my_account',
            merchantDomainName: 'https://my_account.com',
            merchantSecretKey: '12jk3hkj12h3kj1h3kj1h3kj123kh12',
            orderDate: date,
            orderReference: `product__${date}`,
            productName: ['Apple'],
            productPrice: [100],
            currency: 'UAH',
            regularMode: 'daily',
            dateEnd: '24.12.2024',
        });

        expect(mockedAxios.post).toMatchSnapshot();
        expect(response.url).toBe(
            'https://secure.wayforpay.com/page?vkh=regular-some-fake-url-123',
        );
    });

    it('should not generate regular payment url cause of wrong dateEnd format', async () => {
        const date = 1715267902885;
        function testFn() {
            generateRegularPurchaseUrl({
                merchantAccount: 'my_account',
                merchantDomainName: 'https://my_account.com',
                merchantSecretKey: '12jk3hkj12h3kj1h3kj1h3kj123kh12',
                orderDate: date,
                orderReference: `product__${date}`,
                productName: ['Apple'],
                productPrice: [100],
                currency: 'UAH',
                regularMode: 'daily',
                dateEnd: '24-12-2024',
            });
        }

        expect(testFn).toThrowErrorMatchingSnapshot();
    });
});
