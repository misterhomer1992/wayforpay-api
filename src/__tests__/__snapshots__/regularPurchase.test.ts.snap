// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test generateRegularPurchaseUrl should generate regular payment url with dateEnd prop 1`] = `
[MockFunction] {
  "calls": [
    [
      "https://secure.wayforpay.com/pay?behavior=offline",
      FormData {
        Symbol(state): [
          {
            "name": "merchantAccount",
            "value": "my_account",
          },
          {
            "name": "merchantDomainName",
            "value": "https://my_account.com",
          },
          {
            "name": "merchantTransactionSecureType",
            "value": "AUTO",
          },
          {
            "name": "merchantSignature",
            "value": "c2b78f71992bd8f7279cb822756ae2c9",
          },
          {
            "name": "orderReference",
            "value": "product__1715267902885",
          },
          {
            "name": "orderDate",
            "value": "1715267902885",
          },
          {
            "name": "currency",
            "value": "UAH",
          },
          {
            "name": "productName[0]",
            "value": "Apple",
          },
          {
            "name": "productPrice[0]",
            "value": "100",
          },
          {
            "name": "productCount[0]",
            "value": "1",
          },
          {
            "name": "language",
            "value": "EN",
          },
          {
            "name": "orderTimeout",
            "value": "49000",
          },
          {
            "name": "orderLifetime",
            "value": "86400",
          },
          {
            "name": "paymentSystems",
            "value": "card;googlePay;applePay",
          },
          {
            "name": "amount",
            "value": "100",
          },
          {
            "name": "regularBehavior",
            "value": "preset",
          },
          {
            "name": "regularMode",
            "value": "daily",
          },
          {
            "name": "regularAmount",
            "value": "100",
          },
          {
            "name": "regularOn",
            "value": "1",
          },
          {
            "name": "dateEnd",
            "value": "24.12.2024",
          },
        ],
      },
      {
        "headers": {
          "Content-Type": "multipart/form-data",
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`test generateRegularPurchaseUrl should generate regular payment url with regularCount prop 1`] = `
[MockFunction] {
  "calls": [
    [
      "https://secure.wayforpay.com/pay?behavior=offline",
      FormData {
        Symbol(state): [
          {
            "name": "merchantAccount",
            "value": "my_account",
          },
          {
            "name": "merchantDomainName",
            "value": "https://my_account.com",
          },
          {
            "name": "merchantTransactionSecureType",
            "value": "AUTO",
          },
          {
            "name": "merchantSignature",
            "value": "c2b78f71992bd8f7279cb822756ae2c9",
          },
          {
            "name": "orderReference",
            "value": "product__1715267902885",
          },
          {
            "name": "orderDate",
            "value": "1715267902885",
          },
          {
            "name": "currency",
            "value": "UAH",
          },
          {
            "name": "productName[0]",
            "value": "Apple",
          },
          {
            "name": "productPrice[0]",
            "value": "100",
          },
          {
            "name": "productCount[0]",
            "value": "1",
          },
          {
            "name": "language",
            "value": "EN",
          },
          {
            "name": "orderTimeout",
            "value": "49000",
          },
          {
            "name": "orderLifetime",
            "value": "86400",
          },
          {
            "name": "paymentSystems",
            "value": "card;googlePay;applePay",
          },
          {
            "name": "amount",
            "value": "100",
          },
          {
            "name": "regularBehavior",
            "value": "preset",
          },
          {
            "name": "regularMode",
            "value": "daily",
          },
          {
            "name": "regularAmount",
            "value": "100",
          },
          {
            "name": "regularOn",
            "value": "1",
          },
          {
            "name": "regularCount",
            "value": "1",
          },
        ],
      },
      {
        "headers": {
          "Content-Type": "multipart/form-data",
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`test generateRegularPurchaseUrl should not generate regular payment url cause of wrong dateEnd format 1`] = `"Invalid date format. Use DD.MM.YYYY"`;
